modules/swaylock/hm.nix

```diff
@@ -36,13 +45,12 @@ in
         config.stylix.enable
         && config.stylix.targets.swaylock.enable
         && pkgs.stdenv.hostPlatform.isLinux
-
-        # Avoid inadvertently installing the Swaylock package by preventing the
-        # Home Manager module from enabling itself when 'settings != {}' and the
-        # state version is older than 23.05 [1].
+        # Adding `&& config.programs.swaylock.enable` here may lead to infinite
+        # recursion, due to the default value depending on `settings != {}`
+        # when the state version is older than 23.05 [1], and the content of
+        # this module affecting that default.
```

Comment:
Moving the guard to the `options.stylix.targets.swaylock.enable` declaration seems a lot cleaner. Should we simply remove this comment?

Reply:
It's possible someone decides to add that condition in the future, for unrelated reasons. Given that the same pattern works fine in other modules, I thought leaving a comment would avoid this infinite recursion issue being reintroduced by accident.

Reply:
> It's possible someone decides to add that condition in the future, for unrelated reasons. Given that the same pattern works fine in other modules, I thought leaving a comment would avoid this infinite recursion issue being reintroduced by accident.

OK. I guess this comment could be removed once https://github.com/danth/stylix/issues/400 is resolved.